tests/Dropdown.test.js

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import { Dropdown } from "../lib/components";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+const options = ["option 1", "option 2"].map(option => <li key={option}>{option}</li>)
+
+describe("Dropdown", () => {
+  it("should render without error", () => {
+    const { getByText } = render(
+      <Dropdown
+        label="Dropdown"
+      />)
+    expect(getByText("Dropdown")).toBeInTheDocument();
+  });
+
+  it("should show icon when icon component is provided", () => {
+    const { getByTestId } = render(
+      <Dropdown icon={() => <svg data-testid="svg-icon" />} />
+    );
+    expect(getByTestId("svg-icon")).toBeInTheDocument();
+  });
+
+  it("should render all the options when clicking dropdown without error", async () => {
+    const { getByText } = render(
+      <Dropdown label="Dropdown" >
+        {options}
+      </Dropdown>
+    )
+    userEvent.click(getByText("Dropdown"))
+    const listItems = await screen.findAllByRole("listitem")
```

Comment:
Lets query by option text here as it will also help us verify the option text is also rendered.